packages/workspace/src/core/affected-project-graph/locators/workspace-projects.ts

```diff
@@ -4,11 +4,17 @@ export const getTouchedProjects: TouchedProjectLocator = (
   touchedFiles,
   workspaceJson
 ): string[] => {
+  // sort project names with the most nested first,
+  // e.g. ['libs/a/b/c', 'libs/a/b', 'libs/a']
+  const projectKeys = Object.keys(workspaceJson.projects).sort((a, b) =>
+    b.localeCompare(a)
```

Comment:
TargetProjectLocator does similar sorting, but uses project roots. I tink using project root seems more correct here. What do you think?

Reply:
Can we have one function that does the sorting? Or maybe sort `workspaceJson.projects` when we read it?

Reply:
@vsavkin you're right, we should sort by the root value, not the project name (the example given in the comment implies that).

By "sorting when reading" you mean to modify the `readWorkspaceJson` function to return projects descendingly sorted by the root?

Reply:
I have updated the implementation to use the root path instead of the project name. I looked into  `TargetProjectLocator` but `workspaceJson.projects` object seems to be not compatible with `ProjectGraphNodeRecords` used there so I don't know how to re-use that code without extra mapping/refactoring.

Reply:
Thank you very much! I'm going to merge this.